### PR TITLE
Changed colors | Added buttons | Update docs

### PR DESCRIPTION
**Pull request by Roene-JustBetter**
Because we are going to change the color variables in Rapidez/Core, we need to change them here as well.  
See: [#622](https://github.com/rapidez/core/pull/622) and [#657](https://github.com/rapidez/core/pull/657).

## What is changed
- Changed colors for this package. See the pull requests above for more information.
- Added `x-tag` directive.
- Added forms plugin.
- Updated README.
- Updated demo _(Indenting, added buttons & changed colors)_
- Updated components preview.
- Updated screenshots.

I will close https://github.com/rapidez/blade-components/pull/1 as the pull request is from a fork. All feedback in that pull request is addressed here.

### CODE REVIEW

File: resources/views/components/button/index.blade.php
Comment (royduin) [inline resources/views/components/button/index.blade.php:1]: Maybe rename this one to `button.blade.php` just like the other components, see: https://github.com/rapidez/blade-components/pull/7 and all references to it in comments from other files.

File: resources/views/components/button/index.blade.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{{--
+This is the default button if you use the example you get this.
+Example:
+
+```
+<x-rapidez::button>Something</x-rapidez::button>
+```
+--}}
+<x-rapidez::button.base {{ $attributes->twMerge('relative inline-flex items-center justify-center text-white transition bg text-base font-medium rounded min-h-12 py-1.5 px-5 hover:bg-opacity-80 border-b border-black/15 disabled:text-muted disabled:bg-disabled disabled:cursor-not-allowed') }}>
```
Comment (royduin) [inline resources/views/components/button/index.blade.php:9]: - `disabled:bg-disabled` should not exists anymore => `bg-muted`?
- `hover:bg-opacity-80` should we change this to `bg-emphasis`?
- `border-black/15`, do we want/need to be explicit and not use the border colors? Just `border`?

So this becomes: `border bg disabled:bg-muted hover:bg-emphasis`?

File: resources/views/components-preview.blade.php [hunk 2]
```diff
@@ -142,7 +174,20 @@
...
+                <x-rapidez::button>Default</x-rapidez::button>
+                <x-rapidez::button.primary>Primary</x-rapidez::button.primary>
+                <x-rapidez::button.secondary>Secondary</x-rapidez::button.secondary>
+                <x-rapidez::button.outline>Outline</x-rapidez::button.outline>
+                <x-rapidez::button.conversion>Conversion</x-rapidez::button.conversion>
+                <div class="flex flex-col gap-2">
+                    <strong>Slider:</strong>
+                    <div class="flex items-center gap-2">
+                        <x-rapidez::button.slider><</x-rapidez::button.slider>
+                        <x-rapidez::button.slider>></x-rapidez::button.slider>
```
Comment (royduin) [inline resources/views/components-preview.blade.php:187]: This is a bit of a weird syntax, maybe use the encodes? `&lt;` and `&gt;`?

File: resources/views/components/button/conversion.blade.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{{--
+The enhanced button is a variant specifically used for anything conversion related.
```
Comment (royduin) [inline resources/views/components/button/conversion.blade.php:2]: Enhanced => Conversion

File: resources/views/components/button/conversion.blade.php [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+Like purchasing a product or completing the checkout.
+Example:
+
+```
+<x-rapidez::button.enhanced>Something</x-rapidez::button.enhanced>
```
Comment (royduin) [inline resources/views/components/button/conversion.blade.php:7]: Enhanced => Conversion

File: resources/views/components/button/conversion.blade.php [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+```
+--}}
+<x-rapidez::button {{ $attributes->twMerge('bg-conversion text-conversion-text border-b border-black/15') }}>
```
Comment (royduin) [inline resources/views/components/button/conversion.blade.php:10]: `border-black/15` vs `border` vs `bg-conversion/something`? Or should we introduce `conversion-border` just like `conversion-text`?

**Review by royduin — CHANGES_REQUESTED** [review]

File: resources/views/components/button/conversion.blade.php (hunk 5 above)
Comment (Roene-JustBetter) [reply resources/views/components/button/conversion.blade.php]: Removed it here we don't need it as it comes from the `button.blade.php`

File: README.md [hunk 6]
```diff
@@ -16,19 +16,75 @@ We don't provide a input + label combination component as you'll end up with att
...
+- [Button](https://github.com/rapidez/blade-components/blob/master/resources/views/components/button/button.blade.php)
+- [Primary](https://github.com/rapidez/blade-components/blob/master/resources/views/components/button/primary.blade.php)
+- [Secondary](https://github.com/rapidez/blade-components/blob/master/resources/views/components/button/secondary.blade.php)
+- [Outline](https://github.com/rapidez/blade-components/blob/master/resources/views/components/button/outline.blade.php)
+- [Conversion](https://github.com/rapidez/blade-components/blob/master/resources/views/components/button/conversion.blade.php)
+- [Slider](https://github.com/rapidez/blade-components/blob/master/resources/views/components/button/slider.blade.php)
+
+The base button doesn't have any styling; here, we only use the `x-tag`.
+All other buttons use the `button/button`, which contains the basic styling for buttons, such as padding, hover effects, and the disabled state.
+The button variants contain styling for the background and text color. There is only one exception: the `button/slider`, which has more custom styling than the other button variants.
```
Comment (royduin) [inline README.md:31]:
```suggestion
The base button doesn't have any styling; here, we only use the `x-tag`. All other buttons use the `button/button`, which contains the basic styling for buttons, such as padding, hover effects, and the disabled state. The button variants contain styling for the background and text color. There is only one exception: the `button/slider`, which has more custom styling than the other button variants.
```

File: resources/views/components/button/base.blade.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
+{{--
+This is the base for all the buttons. In here we don't need classes.
+If you want to change styling for buttons go to the button/index.
+<x-tag> Is a dynamic tag when a button has a href it will be a <a>,
+when you add a for on the button the button will be a <label>.
+If the button doesn't have a href or label it will be a <button>
```
Comment (royduin) [inline resources/views/components/button/base.blade.php:6]:
```suggestion
- `href` attribute results in an `<a>`
- `for` attribute results in a `<label>`
- Fallback/default is `<button>`
```

File: resources/views/components/button/primary.blade.php [hunk 8]
```diff
@@ -0,0 +1,11 @@
+{{--
+This is the primary button.
+Example:
+
```
Comment (royduin) [inline resources/views/components/button/primary.blade.php:4]:
```suggestion
```

File: resources/views/components/button/secondary.blade.php [hunk 9]
```diff
@@ -0,0 +1,11 @@
+{{--
+This is the secondary button.
+Example:
+
```
Comment (royduin) [inline resources/views/components/button/secondary.blade.php:4]:
```suggestion
```

File: resources/views/components/button/slider.blade.php [hunk 10]
```diff
@@ -0,0 +1,11 @@
+{{--
+This button is used for sliders.
+Example:
+
```
Comment (royduin) [inline resources/views/components/button/slider.blade.php:4]:
```suggestion
```

File: resources/views/components-preview.blade.php [hunk 11]
```diff
@@ -142,7 +174,20 @@
             </div>
 
             <h2 class="font-bold text-lg">Button components</h2>
-            Soon...
+            <div class="grid grid-cols-5 gap-5">
+                <x-rapidez::button>Button/Button</x-rapidez::button>
+                <x-rapidez::button.primary>Button/Primary</x-rapidez::button.primary>
+                <x-rapidez::button.secondary>Button/Secondary</x-rapidez::button.secondary>
+                <x-rapidez::button.outline>Button/Outline</x-rapidez::button.outline>
+                <x-rapidez::button.conversion>Button/Conversion</x-rapidez::button.conversion>
```
Comment (royduin) [inline resources/views/components-preview.blade.php:182]:
```suggestion
                <x-rapidez::button>Button</x-rapidez::button>
                <x-rapidez::button.primary>Primary</x-rapidez::button.primary>
                <x-rapidez::button.secondary>Secondary</x-rapidez::button.secondary>
                <x-rapidez::button.outline>Outline</x-rapidez::button.outline>
                <x-rapidez::button.conversion>Conversion</x-rapidez::button.conversion>
```

**Pull request merged**